src/v3/src/util/formUtils.ts

```diff
@@ -238,7 +225,7 @@ export const getIdpButtonElements = (
     let type = idpObject.type?.toLowerCase() || '';
     let displayName;
 
-    const buttonI18key = SOCIAL_IDP_TYPE_TO_I18KEY[type];
+    const buttonI18key = SOCIAL_IDP_TYPE_TO_I18KEY[type as IdpKey];
```

Comment:
I think we should move this type casting as close to where it's needed as possible. In this case we would move it to the definition of `let type` since we already know here that this is not an accurate type on this line (`type` can also be `''`)

That might look like this (although not certain of the TS syntax here):
```ts
let type = idpObject.type?.toLowerCase() as (IdpKey | undefined) || '';
```

Reply:
This will make it clear when the value is which type. i.e. the left side of the logical OR operator has a union type of `(IdpKey | undefined)` and then we may also look at the right side if the left side had value `undefined`, so the total union type of `type` (a bit confusing 😄 ) assigned to the variable is `IdpKey | ''`.